pyproject.toml

```
@@ -1,15 +1,197 @@
-build-backend = 'setuptools.build_meta'
+requires = [
+    "setuptools>=61.2",
+    "setuptools_scm[toml]>=6.2",
+    "cython>=0.29.30",
+    "oldest-supported-numpy",
+    "extension-helpers",
+]
+build-backend = "setuptools.build_meta"
 
 [tool.setuptools_scm]
 write_to = "photutils/version.py"
+write_to_template = """
+# Note that we need to fall back to the hard-coded version if either
+# setuptools_scm can't be imported or setuptools_scm can't determine the
+# version, so we catch the generic 'Exception'.
+try:
+    from setuptools_scm import get_version
+    version = get_version(root='..', relative_to=__file__)
+except Exception:
+    version = '{version}'
+"""
+
+[tool.setuptools]
+zip-safe = false
+include-package-data = false
+license-files = [
+    "LICENSE.rst",
+]
+
+[tool.setuptools.packages.find]
+namespaces = false
+
+[tool.setuptools.package-data]
+photutils = [
+    "CITATION.rst",
+]
+"photutils.datasets" = [
+    "data/*",
+]
+"photutils.detection.tests" = [
+    "data/*",
+]
+"photutils.isophote.tests" = [
+    "data/*",
+]
+"photutils.psf.tests" = [
+    "data/*",
+]
+
+[tool.pytest.ini_options]
+testpaths = [
+    "photutils",
+    "docs",
+]
+norecursedirs = [
+    "docs[\\/]_build",
+    "docs[\\/]generated",
+    "photutils[\\/]extern",
+]
+astropy_header = true
+doctest_plus = "enabled"
+text_file_format = "rst"
+addopts = "--doctest-rst"
+remote_data_strict = true
+filterwarnings = [
+    "error",
+    "ignore:numpy\\.ufunc size changed:RuntimeWarning",
+    "ignore:numpy\\.ndarray size changed:RuntimeWarning",
+    # numpy 1.24 deprecation from skimage (can remove for skimage >= 0.20)
+    "ignore:`np.bool8` is a deprecated alias for `np.bool_`:DeprecationWarning",
+    # TEMP: remove when matplotlib nightly fixes the row_stack deprecations
+    "ignore:`row_stack` alias is deprecated:DeprecationWarning",
+]
+
+[tool.coverage.run]
+omit = [
+    "photutils/_astropy_init*",
+    "photutils/conftest.py",
+    "photutils/*setup_package*",
+    "photutils/tests/*",
+    "photutils/*/tests/*",
+    "photutils/extern/*",
+    "photutils/version*",
+    "*/photutils/_astropy_init*",
+    "*/photutils/conftest.py",
+    "*/photutils/*setup_package*",
+    "*/photutils/tests/*",
+    "*/photutils/*/tests/*",
+    "*/photutils/extern/*",
+    "*/photutils/version*",
+]
+
+[tool.coverage.report]
+exclude_lines = [
+    "pragma: no cover",
+    "except ImportError",
+    "raise AssertionError",
+    "raise NotImplementedError",
+    "def main\\(.*\\):",
+    "pragma: py{ignore_python_version}",
+    "def _ipython_key_completions_",
+]
 
 [tool.isort]
-    skip_glob = ['photutils/*__init__.py*']
-    known_first_party = ['photutils', 'extension_helpers']
-    use_parentheses = true
+skip_glob = [
+    "photutils/*__init__.py*",
+]
+known_first_party = [
+    "photutils",
+    "extension_helpers",
+]
+use_parentheses = true
+
+[tool.build-sphinx]
+edit-on-github = false
+github-project = 'astropy/photutils'
```

```suggestion
github_project = 'astropy/photutils'
```
This is causing the RTD build to fail.

---

are these still necessary? I've seen usage of `build-sphinx` in `dosc/conf.py`, but as far as I know, the "edit on GitHub" Sphinx extension is deprecated

---

These aren't defined in `setup.cfg` -- I thought you added them.  `edit-on-github` can be removed.  This PR used `github_project` in `docs/conf.py` for resolving issue numbers in the changelog:  https://github.com/astropy/photutils/pull/1488/files#diff-85933aa74a2d66c3e4dcdf7a9ad8397f5a7971080d34ef1108296a7c6b69e7e3R150

Is there a better place to define `astropy/photutils` needed there?